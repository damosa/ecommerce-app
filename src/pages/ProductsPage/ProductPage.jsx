import React, { useEffect, useState } from 'react'
import { getProducts } from './../../apis/products';

const ProductPage = () => {
    const [products, setProducts] = useState(null);

    useEffect(() => {
        getProducts()
        .then(response => setProducts(response.data))
        .catch(error => console.log('error', error))
    }, []);

    return (
        <div>
            {products?.map(product => {
                return(<p>{product.attributes.name}</p>);
                })
            }
        </div>
    )
}

export default ProductPage