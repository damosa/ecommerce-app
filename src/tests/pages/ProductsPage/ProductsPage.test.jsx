import { render } from "@testing-library/react";
import CartProvider from "../../../contexts/CartContext";
import ProductPage from "../../../pages/ProductsPage/ProductPage";
import data from '../../../data.json';

describe('Unit test <ProductsPage /> component', () => {
    beforeAll(() => jest.spyOn(window, 'fetch'));

    it('match to snapshot', () => {
        const { asFragment } = render(<CartProvider><ProductPage /></CartProvider>);
        expect(asFragment()).toMatchSnapshot();
    })

    it('should show a list of products from the API', () => {
        window.fetch.mockResolvedValueOnce({
            ok: true,
            json: async () => data
        });

        render(<CartProvider><ProductPage /></CartProvider>);
        expect(window.fetch).toHaveBeenCalledTimes(1);
        expect(window.fetch).toHaveBeenCalledWith('https://run.mocky.io/v3/729fdd53-f365-49b4-bd55-f7e10bcc9a4b', {"method": "GET"});
    })
})