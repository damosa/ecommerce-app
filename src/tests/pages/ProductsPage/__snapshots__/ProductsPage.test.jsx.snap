// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test <ProductsPage /> component match to snapshot 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
